extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -425,6 +425,34 @@ export class ProjectsController {
 		await vscode.commands.executeCommand(constants.revealFileInOsCommand, vscode.Uri.file(project.projectFilePath));
 	}
 
+	/**
+	 * Opens the .sqlproj file for the given project. Upon update of file, prompts user to
+	 * reload their project.
+	 * @param context a treeItem in a project's hierarchy, to be used to obtain a Project
+	 */
+	public async editProjectFile(context: BaseProjectTreeItem): Promise<void> {
+		const project = this.getProjectFromContext(context);
+
+		try {
+			await vscode.commands.executeCommand(constants.vscodeOpenCommand, vscode.Uri.file(project.projectFilePath));
+			const projFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher(project.projectFilePath);
+
+			projFileWatcher.onDidChange(async (projectFileUri: vscode.Uri) => {
+				const result = await vscode.window.showInformationMessage(constants.refreshProject, constants.yesString, constants.noString);
+
+				if (result === constants.yesString) {
+					// Is there a better way to refresh a project with a changed .sqlproj file than close
```

Comment:
I think `project.readProjFile()` is your best bet, since it'll preserve the tree order.  You may need to reset some of the properties, though.  Probably a good bit of code to add at the beginning of that function.